app.py

```diff
@@ -317,39 +305,33 @@ async def fetch_json(repo_URL, key, file_path, mock=False):
         raise ValueError(str(e))
 
 
-async def get_pipelines(mock:bool = False):
+async def get_pipelines():
     """
     Retrieves the pipelines from the Azure storage API.
 
-    Parameters:
-    - mock (bool): If True, retrieves the pipelines from a mock JSON file. If False, retrieves the pipelines from the Azure storage API.
-
     Returns:
     - list: A list of dictionaries representing the pipelines.
     """
-    if mock:
-        with open("mock_pipeline_json.json", "r+") as f:
-            result_json = json.load(f)
-    else:
-        try:
-            result_json = await azure_storage_api.get_pipeline_info(connection_string, PIPELINE_BLOB_NAME, PIPELINE_VERSION)
-            cipher_suite = Fernet(FERNET_KEY)
-        except PipelineNotFoundError as error:
-            print(error)
-            raise ServerError("Server errror: Pipelines were not found") from error
-    # Get all the api_call function and map them in a dictionary
-    api_call_function = {func.split("from_")[1]: getattr(model_module, func) for func in dir(model_module) if "inference" in func.split("_")}
-    # Get all the inference functions and map them in a dictionary
-    inference_functions = {func: getattr(inference, func) for func in dir(inference) if "process" in func.split("_")}
+    try:
+        app.config["BLOB_CLIENT"] = await azure_storage_api.get_blob_client(connection_string)
+        result_json = await azure_storage_api.get_pipeline_info(app.config["BLOB_CLIENT"], PIPELINE_BLOB_NAME, PIPELINE_VERSION)
+        cipher_suite = Fernet(FERNET_KEY)
+    except (ConnectionStringError, PipelineNotFoundError) as error:
+        print(error)
+        raise ServerError("server errror: could not retrieve the pipelines") from error
+
 
     models = ()
     for model in result_json.get("models"):
         m = Model(
-            api_call_function.get(model.get("api_call_function")),
+            request_function.get(model.get("api_call_function")),
             model.get("model_name"),
+            # To protect sensible data (API key and model endpoint), we crypt them when
```

Comment:
To protect sensitive data (API key and model endpoint), we encrypt it when it's pushed into the blob storage.

Reply:
done